### PR TITLE
(BOLT-28) Ensure Puppet 8 compatibility

### PR DESCRIPTION
**Pull request by LukasAud**
Before this commit, this module did not have Puppet version requirements nor was it clear whether it was compatible with Puppet 8. This PR ensures the minimal necessary work is done to ensure compatibility, and then runs a pdk update on top of that.

Testing was performed through puppetlabs-terraform CI: [8d15fa7](https://github.com/puppetlabs/puppetlabs-terraform/pull/36/commits/8d15fa7ace5287382262931a6b25dbf1d2c02987)

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -5,14 +5,25 @@
...
   "license": "Apache-2.0",
   "source": "https://github.com/puppetlabs/puppetlabs-ruby_plugin_helper",
-  "dependencies": [],
+  "dependencies": [
+
+  ],
+  "operatingsystem_support": [
+    "RedHat based Linux",
+    "Debian based Linux",
+    "Windows"
```
Comment (jordanbreen28) [inline metadata.json:14]: dont think this is valid

File: metadata.json [hunk 2]
```diff
@@ -5,14 +5,25 @@
...
+  ],
   "requirements": [
     {
       "name": "bolt",
       "version_requirement": "< 4.0"
+    },
+    {
+      "name": "puppet",
+      "version_requirement": ">= 6.0.0 < 9.0.0"
```
Comment (jordanbreen28) [inline metadata.json]:
```suggestion
      "version_requirement": ">= 7.0.0 < 9.0.0"
```

File: Gemfile [hunk 3]
```diff
@@ -1,27 +1,79 @@
...
+  gem "parallel_tests", '= 3.12.1',              require: false
+  gem "pry", '~> 0.10',                          require: false
+  gem "simplecov-console", '~> 0.9',             require: false
+  gem "puppet-debugger", '~> 1.0',               require: false
+  gem "rubocop", '~> 1.50.0',                    require: false
+  gem "rubocop-performance", '= 1.16.0',         require: false
+  gem "rubocop-rspec", '= 2.19.0',               require: false
+  gem "rb-readline", '= 0.5.5',                  require: false, platforms: [:mswin, :mingw, :x64_mingw]
+  gem "rexml", '>= 3.0.0', '< 3.2.7',            require: false
+  gem 'puppet-upgrade', git: 'https://github.com/puppetlabs/puppet-upgrade.git'
```
Comment (jordanbreen28) [inline Gemfile]: 👀 
```suggestion
```

**Review by jordanbreen28 — COMMENTED** [review]
Can probably go ahead and set `spec/spec_helper.rb` as unamanged in `.sync.yml` as it isn't actually required in `helper_spec.rb` so just an extra file for no reason 🙂 

I've also just realised there is no ci, so the tests aren't being run..

File: Gemfile (hunk 3 above)
Comment (LukasAud) [reply Gemfile]: oh whoops

File: metadata.json (hunk 2 above)
Comment (LukasAud) [reply metadata.json]: there was no previous requirement so I wanted to give a bit more leeway here, but I guess it make more sense to only list what has been tested

File: metadata.json (hunk 1 above)
Comment (LukasAud) [reply metadata.json:14]: autogenerated by PDK, I would assume it is

Comment (LukasAud) [reply metadata.json:14]: maybe @david22swan can confirm

Comment (jordanbreen28) [reply metadata.json:14]: yeah seems it... 🤔 https://github.com/puppetlabs/pdk/blob/main/lib/pdk/module/metadata.rb#L71C1-L75C15

Comment (jordanbreen28) [reply metadata.json:14]: Don't mind me then!

**Comment by LukasAud** [comment]
> Can probably go ahead and set `spec/spec_helper.rb` as unamanged in `.sync.yml` as it isn't actually required in `helper_spec.rb` so just an extra file for no reason 🙂
> 
> I've also just realised there is no ci, so the tests aren't being run..

`.sync.yml` doesnt exist in this module. I can delete `spec/spec_helper.rb` since it is not particularly useful in this scenario. Testing does not exist for this repo but I have added a link to a terraform PR, which actively uses (and tests that usage) this module.

**Comment by jordanbreen28** [comment]
> `.sync.yml` doesnt exist in this module. I can delete `spec/spec_helper.rb` since it is not particularly useful in this scenario.

It should be added though, that's how we manage cases like the above with PDK. Its better to add the `.sync.yml` and set `spec_helper.rb` to unmanaged or else it will be added again on the next `pdk update`

> Testing does not exist for this repo but I have added a link to a terraform PR, which actively uses (and tests that usage) this module.

Ah yeah i get that, but there is spec tests in this module that should be run, but aren't (https://github.com/puppetlabs/puppetlabs-ruby_plugin_helper/blob/main/spec/helper_spec.rb). This also means rubocop isn't run either

**Review by jordanbreen28 — APPROVED** [review]

**Pull request merged**